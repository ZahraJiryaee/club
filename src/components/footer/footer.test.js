import { shallow } from "enzyme";
import React from "react";
import { shallowToJson } from "enzyme-to-json";

import Footer from "./footer.component";

let wrapper;
// let pathname = "profile";

jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
  useLocation: () => ({
    pathname: "profile",
  }),
}));

beforeEach(() => {
  wrapper = shallow(<Footer />);
});

it("expect to render Footer component with profile path", () => {
  expect(shallowToJson(wrapper)).toMatchSnapshot();

  expect(wrapper.find(".footer-wrapper").prop("style")).toHaveProperty(
    "--footer-indicator-number",
    -2
  );
});
