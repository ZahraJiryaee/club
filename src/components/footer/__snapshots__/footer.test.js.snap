// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render Footer component with profile path 1`] = `
<Fragment>
  <div
    className="footer-wrapper"
    style={
      Object {
        "--footer-indicator-number": -2,
      }
    }
  >
    <div
      className="footer-filler"
    >
      <div
        className="left"
      />
      <div
        className="middle"
      />
      <div
        className="right"
      />
    </div>
    <div
      className="footer-container"
    />
    <div
      className="footer-navigation"
    >
      <ul>
        <li
          className="list active"
        >
          <NavLink
            to="/profile"
          >
            <span
              className="icon"
            >
              <ForwardRef(SvgProfile) />
            </span>
            <span
              className="text"
            >
              پروفایل
            </span>
          </NavLink>
        </li>
        <li
          className="list "
        >
          <NavLink
            to="/shop"
          >
            <span
              className="icon"
            >
              <ForwardRef(SvgShop) />
            </span>
            <span
              className="text"
            >
              فروشگاه
            </span>
          </NavLink>
        </li>
        <li
          className="list "
        >
          <NavLink
            to="/lucky-wheel"
          >
            <span
              className="icon"
            >
              <ForwardRef(SvgWheel) />
            </span>
            <span
              className="text"
            >
              گردونه
            </span>
          </NavLink>
        </li>
        <li
          className="list "
        >
          <NavLink
            to="/games/genre/all"
          >
            <span
              className="icon"
            >
              <ForwardRef(SvgGames) />
            </span>
            <span
              className="text"
            >
              بازی‌ها
            </span>
          </NavLink>
        </li>
        <li
          className="list "
        >
          <NavLink
            to="/leaderboard"
          >
            <span
              className="icon"
            >
              <ForwardRef(SvgLeaderBoard) />
            </span>
            <span
              className="text"
            >
              رده‌بندی
            </span>
          </NavLink>
        </li>
        <div
          className="indicator"
        />
      </ul>
    </div>
  </div>
</Fragment>
`;
